### ISSUE TITLE
ImageSharp sometimes add gray rectangle at the bottom right  

### ISSUE THREAD

**Issue by laitauchocolat** [SixLabors/ImageSharp#2638]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### ImageSharp version

3.1.0

### Other ImageSharp packages and versions

3.0.1

### Environment (Operating system, version and so on)

Windows 11

### .NET Framework version

.NET 6

### Description

On some picture, ImageSharp will add a gray rectangle at the bottom right, 
Original: ![826862_3](https://github.com/SixLabors/ImageSharp/assets/15386570/92539752-5bc7-4cc8-a901-66b93bc6ecc1)
Final: ![square](https://github.com/SixLabors/ImageSharp/assets/15386570/2ff7d034-1c09-4e0d-bc9c-201b79442efc)

### Steps to Reproduce

Download sample projet [c#](https://github.com/SixLabors/Samples/blob/main/ImageSharp/ResizeImage/Program.cs)
Replace fb.jpg with the original picture: 
![826862_3](https://github.com/SixLabors/ImageSharp/assets/15386570/b5d6cac7-3702-46a6-b3ef-e1de73ea75e6)
try it with version 3.0.1, (no gray square)
erease debug + obj and install the latest version  3.1.2 and execute, (gray square)

**Comment by JimBobSquarePants** [comment]
@br3aker would it be possible for you to have a look at this? I thought we’d fixed it previously.

**Comment by omuleanu** [comment]
I got the same thing `net8.0`
![mango](https://github.com/SixLabors/ImageSharp/assets/2010805/f1c47c64-effb-43a5-ac8d-f07df1335209)
![mangoSaved](https://github.com/SixLabors/ImageSharp/assets/2010805/8890d8e0-18f0-49cc-b53e-df5e78e21d3a)

I save it like this:
```
using var fstream = file.OpenReadStream(); // file is IFormFile (asp.net core)

var img = Image.Load(fstream);
await img.SaveAsync(filePath, new JpegEncoder { Quality = 50 });
```

**Comment by JimBobSquarePants** [comment]
Does the issue happen on decode or encode?

>I got the same thing net8.0

What do you mean by this? Same version different target framework?

**Comment by omuleanu** [comment]
@JimBobSquarePants 
version 3.1.2 
net8.0
the image with the gray square is the result of `img.SaveAsync(filePath, new JpegEncoder { Quality = 50 });`
I assume this is encode.

**Comment by JimBobSquarePants** [comment]
OK. Can you please encode the image as Png and let me know whether the result is the same?

**Comment by omuleanu** [comment]
using `await img.SaveAsPngAsync(Path.ChangeExtension(filePath, "png"));` I also get the gray square,
I'm also doing a clone and resize to 300x300 and that one also has the square.

![2](https://github.com/SixLabors/ImageSharp/assets/2010805/40fe91f3-115f-450d-b06c-8e389aaddabd)

**Comment by JimBobSquarePants** [comment]
Decoder then. Will need to figure out the commit that broke it.

**Comment by br3aker** [comment]
Hm, that's strange. I'll take a look this weekend.

**Comment by JimBobSquarePants** [comment]
Thanks @br3aker it was likely something around this. https://github.com/SixLabors/ImageSharp/pull/2516/files#diff-2ffe67fdac3f8753169905e125b8ddb27ec966025580e775f7b07e4c6f2c8d0dR215

**Comment by br3aker** [comment]
> Thanks @br3aker it was likely something around this. [#2516 (files)](https://github.com/SixLabors/ImageSharp/pull/2516/files#diff-2ffe67fdac3f8753169905e125b8ddb27ec966025580e775f7b07e4c6f2c8d0dR215)

That's actually it but fix is rather complex if we don't want to break existing logic with that fast 'DOS images'. Working on it atm.

**Comment by JimBobSquarePants** [comment]
@br3aker I kid you not I was JUST this second writing a comment to see if you wanted me to have a look. 🤪

**Comment by phoebusryan** [comment]
This also happens ion Version 2.1.6. It's not an Version 3-Issue

**Comment by hightechie** [comment]
> This also happens in Version 2.1.6. It's not an Version 3-Issue

I upgraded from 2.1.3 to 2.1.7 due to CVE-2024-27929 and am seeing the gray rectangle on certain images also.

**Comment by JimBobSquarePants** [comment]
The same code is in both v2 and v3

**Comment by phoebusryan** [comment]
Sounds good. Hope he will find a solution, soon :)

**Comment by AndrewGretton** [comment]
Is there any update on this issue? It's negatively impacting us and cropping feels kinda hacky :-)

**Comment by JimBobSquarePants** [comment]
I'll prioritize it. Stay tuned.

**Comment by JimBobSquarePants** [comment]
I've got PRs open for the v3 and v4 branches. It's nearly 1am here now though so I'll see how reviews go in the morning.

**Comment by AndrewGretton** [comment]
Thanks for the amazingly fast turnaround @JimBobSquarePants - appreciate it! Looking forward to a new version appearing on nuget soon.

**Comment by AndrewGretton** [comment]
Hey @JimBobSquarePants - any chance of a new release on nuget 🙏 ? It beats building from source ourselves

**Comment by JimBobSquarePants** [comment]
Working on it! Just gotta get one critical task done first.

**Comment by hightechie** [comment]
@JimBobSquarePants - Is there any chance of this being backported to v2? (we're still dependent on .NET Standard) I just tried upgrading to 2.8 from 2.3 (to address the two CVE's) and the gray rectangle appears with our test images. Thank you kindly if you're able to make this happen.

**Comment by JimBobSquarePants** [comment]
If you open a PR again our 2.1.x release branch I can get one merged. We’re only backporting security releases in-house at the moment to stay on top of maintenance.

**Comment by hightechie** [comment]
> If you open a PR against our 2.1.x release branch I can get one merged. We’re only backporting security releases in-house at the moment to stay on top of maintenance.

I appreciate the quick reply and certainly understand you have to prioritize your time. If I knew how to do this I absolutely would do so, I'm just not sure how to even go about attempting something like this unfortunately. Regardless, thank you for everything you do for the community here.

**Comment by fabercs** [comment]
Hi, I got the same issue in my project .net6.0 and the ImageSharp version is 3.1.4.
I couldn't figure out if this issue fixed in 3.1.4 or for next release, anyone help?

**Comment by JimBobSquarePants** [comment]
> Hi, I got the same issue in my project .net6.0 and the ImageSharp version is 3.1.4. I couldn't figure out if this issue fixed in 3.1.4 or for next release, anyone help?

The fix has been released. If you are still seeing an issue, please raise a separate ticket with the image in question.

**Comment by phoebusryan** [comment]
It still does not work with 2.1.9 and .net Framework 4.6.2. Example images here:

![70922](https://github.com/user-attachments/assets/01fb9f9d-45fd-44a6-8741-15efbdf2fb78)
![70968](https://github.com/user-attachments/assets/eb12ee92-7698-4063-8c71-4a7c46e1925d)
![70973_2](https://github.com/user-attachments/assets/81bb8e9e-253f-427b-b5a4-c20326ef2454)
![70974_2](https://github.com/user-attachments/assets/90bc67fc-fd21-4221-8c8e-a662f458cc5b)
![70977_2](https://github.com/user-attachments/assets/af17bdf4-e878-4460-ba30-85938942c264)

**Comment by antonfirsov** [comment]
2.1.* is unmaintained, only security-critical fixes are being backported.

**Comment by phoebusryan** [comment]
So this issue will never fixed in 2.1.x?  Unfortunately, we cannot use version 3 because we do not (yet) use .net core

**Comment by JimBobSquarePants** [comment]
> If you open a PR against our 2.1.x release branch I can get one merged. We’re only backporting security releases in-house at the moment to stay on top of maintenance.

**Comment by Visual-Vincent** [comment]
I have opened a pull request (#2868) that backports this to 2.1.x.

**Comment by Visual-Vincent** [comment]
The PR was finalized and merged by the maintainers - Thank you for the assistance!

These changes have now been released as part of [v2.1.10](https://github.com/SixLabors/ImageSharp/releases/tag/v2.1.10).
